src/drivers/barometer/bmp581/bmp581.cpp

```diff
@@ -0,0 +1,703 @@
+	perf_free(_comms_errors);
+
+	delete _interface;
+}
+
+int BMP581::init()
+{
+	if(!soft_reset()) {
+		PX4_DEBUG("failed to reset baro during init");
+		return -EIO;
+	}
+
+	_chip_id = _interface->get_reg(BMP5_REG_CHIP_ID_ADDR);
+
+	if((_chip_id != BMP581_CHIP_ID_PRIM) && (_chip_id != BMP581_CHIP_ID_SEC)) {
+		PX4_WARN("id of your baro is not: 0x%02x or 0x%02x", BMP581_CHIP_ID_PRIM, BMP581_CHIP_ID_SEC);
+		return -EIO;
+	}
+
+	_chip_rev_id = _interface->get_reg(BMP5_REG_REV_ID_ADDR);
+
+	if(!set_config()) {
+		PX4_WARN("failed to set_config");
+		return -EIO;
+	}
+
+	start();
+
+	return OK;
+}
+
+void BMP581::print_status()
+{
+	I2CSPIDriverBase::print_status();
+	printf("chip id: 0x%02x rev id: 0x%02x\n", _chip_id, _chip_rev_id);
+	perf_print_counter(_sample_perf);
+	perf_print_counter(_measure_perf);
+	perf_print_counter(_comms_errors);
+	printf("measurement interval: %u us \n", _measure_interval);
+}
+
+void BMP581::start()
+{
+	_collect_phase = false;
+
+	ScheduleOnInterval(_measure_interval, _measure_interval * 3);
+}
+
+void BMP581::RunImpl()
+{
+
+	if(_collect_phase) {
+		collect();
+	}
+
+	measure();
+}
+
+int BMP581::measure()
+{
+	_collect_phase = true;
+
+	perf_begin(_measure_perf);
+
+	/* start measurement */
+	if(!set_power_mode(BMP5_POWERMODE_FORCED)) {
+		PX4_DEBUG("failed to set power mode");
+		perf_count(_comms_errors);
+		perf_cancel(_measure_perf);
+		return -EIO;
+	}
+
+	perf_end(_measure_perf);
+
+	return OK;
+}
+
+int BMP581::collect()
+{
+	_collect_phase = false;
+
+	bmp5_sensor_data data{};
+
+	uint8_t int_status;
+	int ret;
+
+	perf_begin(_sample_perf);
+
+	/* this should be fairly close to the end of the conversion, so the best approximation of the time */
+	const hrt_abstime timestamp_sample = hrt_absolute_time();
+
+	ret = get_interrupt_status(&int_status);
+
+	if(ret) {
+		if(int_status & BMP5_INT_ASSERTED_DRDY) {
+			if(!get_sensor_data(&data)) {
+				perf_count(_comms_errors);
+				perf_cancel(_sample_perf);
+				return -EIO;
+			}
+		}
+	}
+
+	//publish
+	sensor_baro_s sensor_baro{};
+	sensor_baro.timestamp_sample = timestamp_sample;
+	sensor_baro.device_id = _interface->get_device_id();
+	sensor_baro.pressure = data.pressure;
+	sensor_baro.temperature = data.temperature;
+	sensor_baro.error_count = perf_event_count(_comms_errors);
+	sensor_baro.timestamp = hrt_absolute_time();
+	_sensor_baro_pub.publish(sensor_baro);
+
+	perf_end(_sample_perf);
+
+	return OK;
+}
+
+/*!
+ * @brief This API performs the soft reset of the sensor
+ *
+ * https://github.com/boschsensortec/BMP5-Sensor-API/blob/master/bmp5.c
+ */
+bool BMP581::soft_reset()
+{
+	bool result = false;
+	 uint8_t status;
+	int ret;
+
+	if(intf == BMP5_SPI_INTF) {
+		/* Performing a single read via SPI of registers,
+		* e.g. registers CHIP_ID, before the actual
+		* SPI communication with the device.
+		*/
+		status = _interface->get_reg(BMP5_REG_CHIP_ID_ADDR);
+	}
+
+	ret = _interface->set_reg(BMP5_SOFT_RESET_CMD, BMP5_REG_CMD_ADDR);
+
+	if(ret == OK) {
```

Comment:
Check the failed condition and return early rather than nesting. Apply the pattern everywhere else in the driver.
```
if (ret != PX4_OK) {
    return false;
}
```